### PR TITLE
Release Notes update for 2021.2

### PR DESCRIPTION
**Pull request by ValentinaKats**
Release Notes update for 2021.2 and typo fix in 2021.1 Release Notes

### CODE REVIEW

**Comment by capatober** [comment]
Do we plan to add it to `CHANGES.rst` file now or it will be done separately?

**Comment by ValentinaKats** [comment]
> Do we plan to add it to `CHANGES.rst` file now or it will be done separately?

It will be done separately after the review of this PR is finished.

File: documentation/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,52 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+Changes to Existing Features
+-----------------------------
+- Improved performance of the ``sort`` and ``stable_sort`` algorithms when using Radix sort.
+- Improved performance of the linear_congruential_engine RNG engine (including ``minstd_rand``, ``minstd_rand0``, ``minstd_rand_vec``, ``minstd_rand0_vec`` predefined engines).
+- Improved performance of the following algorithms: ``count``, ``count_if``, ``is_partitioned``, ``lexicographical_compare``, ``max_element``, ``min_element``, ``minmax_element``, ``reduce``, ``transform_reduce``.
+
+Fixed Issues
+------------
+- Fixed runtime errors on CPU building with the -O0 option for the following algorithms: ``find_end``, ``search``, ``search_n``.
+- Removed unused parameters names in most of the functions that were producing warnings. In most cases the parameters were needed for function overloading.
```
Comment (dsobolev-dev) [inline documentation/release_notes.rst:29]: Seems the description is too verbose. Let's leave only what affects a oneDPL user.

```suggestion
- Fixed unused parameter warnings.
```

File: documentation/release_notes.rst [hunk 2]
```diff
@@ -8,6 +8,52 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+
+Known Issues and Limitations
+-----------------------------
+- ``exclusive_scan`` and ``transform_exclusive_scan`` algorithms may provide wrong results with ``unseq`` and ``par_unseq`` policies
```
Comment (dsobolev-dev) [inline documentation/release_notes.rst:33]: Align with the other comments. 
```suggestion
- ``exclusive_scan`` and ``transform_exclusive_scan`` algorithms may provide wrong results with vector execution policies
```

File: documentation/release_notes.rst [hunk 3]
```diff
@@ -8,6 +8,52 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+
+Fixed Issues
+------------
+- Fixed runtime errors on CPU building with the -O0 option for the following algorithms: ``find_end``, ``search``, ``search_n``.
+- Removed unused parameters names in most of the functions that were producing warnings. In most cases the parameters were needed for function overloading.
+
+Known Issues and Limitations
+-----------------------------
+- ``exclusive_scan`` and ``transform_exclusive_scan`` algorithms may provide wrong results with par and par_unseq policies
+  when using GCC 10 and building a program in debug mode.
```
[the remark this answers is not in the document]
Comment (dsobolev-dev) [reply documentation/release_notes.rst:36]: The issue occurs with both 10.2 and 10.1.
I imply -O0 option when saying debug mode. I think is it confusing and should be changed here and in the other places.

[the remark this answers is not in the document]
Comment (dsobolev-dev) [reply documentation/release_notes.rst:36]:
```suggestion
  when building a program with GCC 10 and using -O0 option.
```

File: documentation/release_notes.rst [hunk 4]
```diff
@@ -34,7 +82,7 @@ Known Issues and Limitations
   with some compilers including GCC 7 and earlier. Instead of this directive, explicitly use
   ``oneapi::dpl`` namespace, or create a namespace alias.
 - The ``partial_sort_copy``, ``sort`` and ``stable_sort`` algorithms are prone to ``CL_BUILD_PROGRAM_FAILURE``
```
Comment (dsobolev-dev) [inline documentation/release_notes.rst:84]: This is not relevant anymore. Let's remove it.

File: documentation/release_notes.rst [hunk 5]
```diff
@@ -8,6 +8,54 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+Fixed Issues
+------------
+- Fixed runtime errors on CPU building with the -O0 option for the following algorithms: ``find_end``, ``search``, ``search_n``.
+- Removed unused parameters names in most of the functions that were producing warnings. In most cases the parameters were needed for function overloading.
+
+Known Issues and Limitations
+-----------------------------
+- ``exclusive_scan`` and ``transform_exclusive_scan`` algorithms may provide wrong results with ``unseq`` and ``par_unseq`` policies
+  when using GCC 10 and building a program in debug mode.
+- Some algorithms may hang when a program is built in debug mode, executed on GPU devices and large number of elements is to be processed.
```
Comment (dsobolev-dev) [inline documentation/release_notes.rst:37]:
```suggestion
- Some algorithms may hang when a program is built with -O0 option, executed on GPU devices and large number of elements is to be processed.
```

File: documentation/release_notes.rst [hunk 6]
```diff
@@ -34,7 +82,7 @@ Known Issues and Limitations
   with some compilers including GCC 7 and earlier. Instead of this directive, explicitly use
   ``oneapi::dpl`` namespace, or create a namespace alias.
 - The ``partial_sort_copy``, ``sort`` and ``stable_sort`` algorithms are prone to ``CL_BUILD_PROGRAM_FAILURE``
-  when using Radix sort [#fnote1]_ in debug mode on oneAPI CPU devices.
+  when using Radix sort [#fnote1]_ in debug mode on CPU devices.
```
Comment (dsobolev-dev) [inline documentation/release_notes.rst:85]:
```suggestion
  when a program uses Radix sort [#fnote1]_, is built with -O0 option and executed on CPU devices.
```

File: documentation/release_notes.rst [hunk 7]
```diff
@@ -8,6 +8,54 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+------------
+-  Added support of parallel, vector and DPC++ execution policies for the following algorithms: ``shift_left``, ``shift_right``.
+-  Added the Range-based versions of the following algorithms: ``sort``, ``stable_sort``, ``merge``.
+-  Added non-blocking versions of the following algorithms: ``copy``, ``fill``, ``for_each``, ``reduce``, ``sort``, ``transform``, ``transform_reduce``. These algorithms are declared in ``oneapi::dpl::experimental`` namespace with suffix _async and implemented only for DPC++ policies. In order to make these algorithms available the ``<oneapi/dpl/async>`` header should be included. Use of the non-blocking API requires C++11.
+-	 Utility function ``wait_for_all`` enables waiting for completion of an arbitrary number of events.
+-  Added the ``ONEDPL_USE_PREDEFINED_POLICIES`` macro that can disable predefined policies objects and make functions without arguments if needed.
+
+Changes to Existing Features
+-----------------------------
+- Improved performance of the ``sort`` and ``stable_sort`` algorithms when using Radix sort.
```
Comment (dsobolev-dev) [inline documentation/release_notes.rst:24]:
```suggestion
- Improved performance of the ``sort`` and ``stable_sort`` algorithms when using Radix sort [#fnote1]_.
```

File: documentation/release_notes.rst [hunk 8]
```diff
@@ -8,6 +8,54 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
 
+New in 2021.2
+=============
+
+New Features
+------------
+-  Added support of parallel, vector and DPC++ execution policies for the following algorithms: ``shift_left``, ``shift_right``.
+-  Added the Range-based versions of the following algorithms: ``sort``, ``stable_sort``, ``merge``.
+-  Added non-blocking versions of the following algorithms: ``copy``, ``fill``, ``for_each``, ``reduce``, ``sort``, ``transform``, ``transform_reduce``. These algorithms are declared in ``oneapi::dpl::experimental`` namespace with suffix _async and implemented only for DPC++ policies. In order to make these algorithms available the ``<oneapi/dpl/async>`` header should be included. Use of the non-blocking API requires C++11.
+-	 Utility function ``wait_for_all`` enables waiting for completion of an arbitrary number of events.
```
Comment (capatober) [inline documentation/release_notes.rst:19]: Extra spaces at he beginning
```suggestion
-  Utility function ``wait_for_all`` enables waiting for completion of an arbitrary number of events.
```

File: documentation/release_notes.rst [hunk 9]
```diff
@@ -8,6 +8,54 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+-  Added the ``ONEDPL_USE_PREDEFINED_POLICIES`` macro, which enables predefined policy objects and ``make_device_policy``, ``make_fpga_policy`` functions without arguments. It is turned on by default.
+
+Changes to Existing Features
+-----------------------------
+- Improved performance of the ``sort`` and ``stable_sort`` algorithms when using Radix sort [#fnote1]_.
+- Improved performance of the linear_congruential_engine RNG engine (including ``minstd_rand``, ``minstd_rand0``, ``minstd_rand_vec``, ``minstd_rand0_vec`` predefined engines).
+- Improved performance of the following algorithms: ``count``, ``count_if``, ``is_partitioned``, ``lexicographical_compare``, ``max_element``, ``min_element``, ``minmax_element``, ``reduce``, ``transform_reduce``.
```
Comment (capatober) [inline documentation/release_notes.rst:26]: I think we can merge 24 and 26 lines into one line

Comment (ValentinaKats) [reply documentation/release_notes.rst:26]:
```suggestion
- Improved performance of the following algorithms: ``count``, ``count_if``, ``is_partitioned``, ``lexicographical_compare``, ``max_element``, ``min_element``, ``minmax_element``, ``reduce``, ``transform_reduce``, and ``sort``, ``stable_sort`` when using Radix sort [#fnote1]_.
```
Something like this?

Comment (capatober) [reply documentation/release_notes.rst:26]: Yes. something like that

File: documentation/release_notes.rst [hunk 10]
```diff
@@ -8,6 +8,53 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+
+Fixed Issues
+------------
+- Fixed runtime errors occurring with``find_end``, ``search``, ``search_n`` algorithms when a program is built with -O0 option and executed on CPU devices.
+- Fixed the majority of unused parameter warnings.
+
+Known Issues and Limitations
+-----------------------------
+- ``exclusive_scan`` and ``transform_exclusive_scan`` algorithms may provide wrong results with vector execution policies
+  when using GCC 10 and building a program in debug mode.
```
Comment (dsobolev-dev) [inline documentation/release_notes.rst:35]: Debug mode -> -O0 option
```suggestion
  when building a program with GCC 10 and using -O0 option.
```

**Review by capatober — APPROVED** [review]

**Pull request merged**